DataFormats/HcalRecHit/src/HBHERecHit.cc

```diff
@@ -62,3 +62,22 @@ void HBHERecHit::getMergedIds(std::vector<HcalDetId>* ids) const
         }
     }
 }
+
+
+HcalDetId HBHERecHit::idFront() const {
+  if (isMerged()) {
+    HcalDetId myId(id());
+    if (auxPhase1_ & (1U << HBHERecHitAuxSetter::OFF_COMBINED)) {
+      const unsigned nMerged = CaloRecHitAuxSetter::getField(auxPhase1_,
+							     HBHERecHitAuxSetter::MASK_NSAMPLES,
+							     HBHERecHitAuxSetter::OFF_NSAMPLES);
+      if (nMerged > 0) {
+	const unsigned depth = CaloRecHitAuxSetter::getField(auxHBHE_, 0xf, 0);
+	myId = HcalDetId(myId.subdet(), myId.ieta(), myId.iphi(), depth);
```

Comment:
I understand it will get the depth of the front one of the merged cell - if depths 1,2,3 are merged as depth 1 this will give 1 for depth 1; if depths 4,5,6 are merged as depth 2, this will give 4 for depth 2.